### PR TITLE
Replace most generated code in SyntaxCollections.swift by default protocol implementations

### PR DESCRIPTION
**Pull request by ahoppen**
This has two advantages.

1. It reduces the amout of code in swift-syntax
2. It actually makes all the functions defined in every `SyntaxCollection` accessible for existentials or generics over `SyntaxCollection`.

### CODE REVIEW

**Review by DougGregor — APPROVED** [review]
This much code deletion brings happy tears

File: Sources/SwiftSyntaxBuilder/generated/BuildableCollectionNodes.swift [hunk 1]
```diff
@@ -173,7 +173,7 @@ extension ExprListSyntax: ExpressibleByArrayLiteral {
   public init(_ elements: [ExprSyntaxProtocol]) {
     self = ExprListSyntax(elements.map {
       ExprSyntax(fromProtocol: $0)
-    })
+    } as [ExprSyntax])
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply Sources/SwiftSyntaxBuilder/generated/BuildableCollectionNodes.swift:176]: I think it’s something to do with the initializer being moved to an extension and thus it doesn’t have higher precedence than this initializer that that takes `[ExprSyntaxProtocol]` anymore. But I haven’t reduced it to a minimal reproducer.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by kimdv — APPROVED** [review]
😍

**Comment by ahoppen** [comment]
@swift-ci Please test

**Pull request merged**